rust/tw_hash/src/hash_array.rs

```diff
@@ -44,6 +75,28 @@ impl<const N: usize> Hash<N> {
     }
 }
 
+/// This is a [`Hash::split`] helper that ensures that `L + R == N` at compile time.
+/// Assertion example:
+/// ```rust(ignore)
+/// let hash = H256::default();
+/// let (left, right): (H128, H160) = hash.split();
+///
+/// // output:
+/// // error[E0080]: evaluation of `hash_array::AssertSplit::<16, 20, 32>::EQ` failed
+/// //   --> tw_hash/src/hash_array.rs:67:41
+/// //    |
+/// // 67 |     pub const EQ: usize = (R + L - N) + (N - (R + L));
+/// //    |                                         ^^^^^^^^^^^^^ attempt to compute `32_usize - 36_usize`, which would overflow
+/// ```
+///
+/// TODO remove once [feature(generic_const_exprs)](https://github.com/rust-lang/rust/issues/76560) is stable.
+struct AssertSplit<const L: usize, const R: usize, const N: usize>;
+
+/// cbindgen:ignore
+impl<const L: usize, const R: usize, const N: usize> AssertSplit<L, R, N> {
+    pub const VALID: usize = (R + L - N) + (N - (R + L));
```

Comment:
How does this ensure that `L + R == N`? If I understand correctly, `VALID` should be `0` in such a case but that does not seem to be evaluated anywhere?